SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.cpp

```diff
@@ -254,21 +255,58 @@ void SSpatialOutputLog::StartPollTimer(const FString& LogFilePath)
 	});
 }
 
+void SSpatialOutputLog::FormatAndPrintRawErrorLine(const FString& LogLine)
+{
+	const FRegexPattern ErrorPattern = FRegexPattern(TEXT("level=(.*) msg=(.*) code=(.*) code_string=(.*) error=(.*) stack=(.*)"));
+	FRegexMatcher ErrorMatcher(ErrorPattern, LogLine);
+
+	if (!ErrorMatcher.FindNext())
+	{
+		UE_LOG(LogSpatialOutputLog, Error, TEXT("Failed to parse log line: %s"), *LogLine);
+		return;
+	}
+
+	FString ErrorLevelText = ErrorMatcher.GetCaptureGroup(1);
+	FString Message = ErrorMatcher.GetCaptureGroup(2);
+	FString ErrorCode = ErrorMatcher.GetCaptureGroup(3);
+	FString ErrorCodeString = ErrorMatcher.GetCaptureGroup(4);
+	FString ErrorMessage = ErrorMatcher.GetCaptureGroup(5);
+	FString Stack = ErrorMatcher.GetCaptureGroup(6);
+
+	// The stack message comes with double escaped characters.
+	Stack = Stack.ReplaceEscapedCharWithChar();
+
+	// Format the log message to be easy to read.
+	FString LogMessage = FString::Printf(TEXT("%s \n Code: %s \n Code String: %s \n Error: %s \n Stack: %s"), *Message, *ErrorCode, *ErrorCodeString, *ErrorMessage, *Stack);
+
+	ELogVerbosity::Type LogVerbosity = ELogVerbosity::Error;
+
+	// There are currently no logger names included in the errors from Spatial Service and so we always give it this LogCategory.
+	FString LogCategory = TEXT("SpatialService");
```

Comment:
Just use an FName directly?

Reply:
Yeah thought about that, there was a chance we were going to get better LogCategories from the Spatial Service so I left it in, doesn't look like that will happen anymore though so we can safely remove this.